etcdserver/api/v3rpc/util.go

```diff
@@ -116,3 +117,15 @@ func isClientCtxErr(ctxErr error, err error) bool {
 	}
 	return false
 }
+
+// in v3.4, learner is allowed to serve serializable read and endpoint status
+func isRPCEnabledForLearner(req interface{}) bool {
```

Comment:
Do we need anything for endpoint health, or for the configuration APIs?

Reply:
Good point. Currently endpoint health use linearizable read as part of the health check. By definition, learner is unhealthy...

Reply:
I think endpoint health handler can be addressed in a different PR (since it's just HTTP endpoint as proxy to quorum reads). This PR can just handle the gRPC layer.